Source/WebKit/UIProcess/API/glib/WebKitImage.cpp

```diff
@@ -0,0 +1,381 @@
+static void webkit_image_gloadable_icon_interface_init(GLoadableIconIface*);
+
+struct _WebKitImagePrivate {
+    int width;
+    int height;
+    unsigned stride;
+    CString type;
+    WebKitImageCreateBytesFunction createBytesFn;
+
+    GRefPtr<GBytes> bytes;
+};
+
+WEBKIT_DEFINE_FINAL_TYPE_WITH_CODE(
+    WebKitImage, webkit_image, G_TYPE_OBJECT, GObject,
+    G_IMPLEMENT_INTERFACE(G_TYPE_ICON, webkit_image_gicon_interface_init)
+    G_IMPLEMENT_INTERFACE(G_TYPE_LOADABLE_ICON, webkit_image_gloadable_icon_interface_init))
+
+
+enum {
+    PROP_0,
+
+    PROP_WIDTH,
+    PROP_HEIGHT,
+    PROP_STRIDE,
+    PROP_TYPE,
+
+    N_PROPERTIES
+};
+
+static std::array<GParamSpec*, N_PROPERTIES> sObjProperties;
+
+static void webkitImageSetProperty(GObject* object, guint propId, const GValue* value, GParamSpec* paramSpec)
+{
+    auto* image = WEBKIT_IMAGE(object);
+
+    switch (propId) {
+    case PROP_WIDTH:
+        image->priv->width = g_value_get_int(value);
+        break;
+    case PROP_HEIGHT:
+        image->priv->height = g_value_get_int(value);
+        break;
+    case PROP_STRIDE:
+        image->priv->stride = g_value_get_uint(value);
+        break;
+    case PROP_TYPE:
+        image->priv->type = CString(g_value_get_string(value));
+        break;
+    default:
+        G_OBJECT_WARN_INVALID_PROPERTY_ID(object, propId, paramSpec);
+    }
+}
+
+static void webkitImageGetProperty(GObject* object, guint propId, GValue* value, GParamSpec* paramSpec)
+{
+    auto* image = WEBKIT_IMAGE(object);
+
+    switch (propId) {
+    case PROP_WIDTH:
+        g_value_set_int(value, webkit_image_get_width(image));
+        break;
+    case PROP_HEIGHT:
+        g_value_set_int(value, webkit_image_get_height(image));
+        break;
+    case PROP_STRIDE:
+        g_value_set_uint(value, webkit_image_get_stride(image));
+        break;
+    case PROP_TYPE:
+        g_value_set_string(value, webkit_image_get_image_type(image));
+        break;
+    default:
+        G_OBJECT_WARN_INVALID_PROPERTY_ID(object, propId, paramSpec);
+    }
+}
+
+static void webkit_image_class_init(WebKitImageClass* imageClass)
+{
+    GObjectClass* objectClass = G_OBJECT_CLASS(imageClass);
+    objectClass->set_property = webkitImageSetProperty;
+    objectClass->get_property = webkitImageGetProperty;
+
+    /**
+     * WebKitImage:width:
+     *
+     * The image width
+     */
+    sObjProperties[PROP_WIDTH] =
+    g_param_spec_int(
+        "width",
+        nullptr, nullptr,
+        0, G_MAXINT, 0,
+        static_cast<GParamFlags>(WEBKIT_PARAM_READWRITE | G_PARAM_CONSTRUCT_ONLY));
+
+    /**
+     * WebKitImage:height:
+     *
+     * The image height
+     */
+    sObjProperties[PROP_HEIGHT] =
+    g_param_spec_int(
+        "height",
+        nullptr, nullptr,
+        0, G_MAXINT, 0,
+        static_cast<GParamFlags>(WEBKIT_PARAM_READWRITE | G_PARAM_CONSTRUCT_ONLY));
+
+    /**
+     * WebKitImage:stride:
+     *
+     * The image stride
```

Comment:
It would not hurt to have indicate the units here, and throw in a bit more of information for those using the API that may not know just from the word “stride” what this represents:


```suggestion
     * The image stride, in bytes. This value indicates the amount of
     * memory occupied by each row of pixels in the image. Note that
     * the stride may be larger than the image width multiplied by the
     * amount of bytes used to represent each pixel.
```